### ISSUE TITLE
Random seed

### ISSUE THREAD

**Issue by danheck** [mpt-network/mptcomparison#4]
I think it would be good to use a random seed at the top of the script to ensure reproducibility. 

* TreeBUGS uses the default method of setting a seed, so this would suffice: `set.seed(1234)` 
* What about MPTinR in parallel?

**Comment by singmann** [comment]
In this script, we actually do not use the `MPTinR` parallel computation capabilities, but create a [`parallel` package](https://stat.ethz.ch/R-manual/R-devel/library/parallel/doc/parallel.pdf) `"localhost"` cluster via `parallel::makeCluster` (I believe this is equivalent to `parallel::makePSOCKcluster`). Thus, we would only need to ensure to set the appropriate seed on the cluster nodes. I believe this can be done via `parallel::clusterSetRNGStream`.

**Comment by singmann** [comment]
I have just added the possibility of using a `seed` to the `MPTinR` part:  9b2ff36
Besides using `parallel::clusterSetRNGStream`, I also needed to change `clusterApplyLB` to `clusterApply`.

Now, to use a seed, one needs to pass an integer `seed` to `mpt_mptinr()` (in addition to using `set.seed`):
```
seed <- 666
set.seed(seed)
res_mptinr <- mpt_mptinr(dataset = DATA_FILE, data = data, model = EQN_FILE,
                         col_id = COL_ID, col_condition = COL_CONDITION, 
                         seed = seed)
```
As can be seen in the following, this then perfectly replicates the results:

```
seed <- 666
set.seed(seed)
res_mptinr2 <- mpt_mptinr(dataset = DATA_FILE, data = data, model = EQN_FILE,
                         col_id = COL_ID, col_condition = COL_CONDITION, 
                         seed = seed)
all_equal(res_mptinr$est_group, res_mptinr2$est_group)
# [1] TRUE
all.equal(as.data.frame(res_mptinr$est_indiv[[1]]), 
          as.data.frame(res_mptinr2$est_indiv[[1]]))
# [1] TRUE
all.equal(as.data.frame(res_mptinr$est_indiv[[2]]), 
          as.data.frame(res_mptinr2$est_indiv[[2]]))
# [1] TRUE
all.equal(as.data.frame(res_mptinr$gof_indiv[[1]]), 
          as.data.frame(res_mptinr$gof_indiv[[1]]))
# [1] TRUE
all.equal(as.data.frame(res_mptinr$gof_indiv[[2]]), 
          as.data.frame(res_mptinr$gof_indiv[[2]]))
# [1] TRUE
```

Note that I did not add this to any of the scripts. I just added the possibility.

**Comment by danheck** [comment]
I was just working on a similar idea. What about the following solution: we start the MPTinR cluster with :

`clusterSetRNGStream(cl, iseed = sample.int(.Machine$integer.max, 1))`

then, it is sufficient to use`set.seed(123)` at the top of the script, so there are not additional arguments...

Edit: This is the way how TreeBUGS does it.

**Comment by singmann** [comment]
Sounds like an excellent idea. To avoid us both working on it at the same time. I just leave it to you to make changes now. Unless I hear otherwise.

**Comment by danheck** [comment]
ok, I will change this accordingly.

**Comment by danheck** [comment]
fixed in 7d0ab75ffbe6a7bc1d4d176f9f869405ee5b3b6a
